lib/user.rb

```diff
@@ -0,0 +1,29 @@
+class User
+  attr_reader :name, :jokes, :id
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(new_joke)
+    @jokes << new_joke
+  end
+
+  def tell(audience, joke)
+    audience.jokes << joke
```

Comment:
This works, but you could also use the `learn` method instead of shoveling:

```
audience.learn(joke)
```

This way, one user is not responsible for changing the state of another user - they just give that user enough information to change their own state.